pkg/asset-manager-utils/contracts/RewardsAssetManager.sol

```diff
@@ -226,21 +226,10 @@ abstract contract RewardsAssetManager is IAssetManager {
         poolManaged = aum;
     }
 
-    /**
-     * @notice Checks whether the pool should rebalance upon adding or removing `delta` cash.
-     */
-    function shouldRebalance(uint256 delta, bool positiveDelta) external view override returns (bool) {
-        (uint256 poolCash, uint256 poolManaged) = _getPoolBalances(_getAUM());
-
-        // We calculate the value of cash held by the pool after the join/exit
-        uint256 updatedCash = positiveDelta ? poolCash.add(delta) : poolCash.sub(delta);
-        return _shouldRebalance(updatedCash, poolManaged);
-    }
-
     /**
      * @notice Determines whether the pool should rebalance given the provided balances
      */
-    function _shouldRebalance(uint256 cash, uint256 managed) private view returns (bool) {
+    function shouldRebalance(uint256 cash, uint256 managed) public view override returns (bool) {
         uint256 investedPercentage = cash.mul(FixedPoint.ONE).divDown(cash + managed);
```

Comment:
Given that this is now `public` and the amounts are caller-supplied, we shouldn't make assumptions on them and use SafeMath in the `cash + managed` addition.

Reply:
Fair enough. My thoughts are that a user could provide wildly incorrect values while still not causing an overflow and even if they do then it doesn't affect AM logic. That said it doesn't really hurt.

Reply:
Agreed, but it feels a bit weird that we'd knowlingly process incorrect data and return an invalid result :shrug: